pytest_catchlog.py

```diff
@@ -11,6 +12,29 @@
 
 __version__ = '1.2.0'
 
+TERMINAL = py.io.TerminalWriter(sys.stderr)  # pylint: disable=no-member
+CONSOLEHANDLER = logging.StreamHandler(TERMINAL)
+# Add the handler to logging
+logging.root.addHandler(CONSOLEHANDLER)
+# The root logging should have the lowest logging level to allow all messages
+# to be "passed" to the handlers
+logging.root.setLevel(1)
+
+
+class Hooks(object):  # pylint: disable=too-few-public-methods
+    """
+    Class to add new hooks to pytest
+    """
+
+    def pytest_register_logging_levels(self):
```

Comment:
> I don't think you can get a fixed formula to account for user defined levels...

Why not? User level are anyway somewhere between NOTSET and CRITICAL, more likely between NOTSET and DEBUG. So we could just take it into consideration, i.e. scale the range properly, for example:
```
ERROR   (40) default
WARNING (30) -v 
INFO    (20) -vv
DEBUG   (10) -vvv
CUSTOM  (5)  -vvvv
CUSTOM  (2)  -vvvvv
```